src/compiler/watchUtilities.ts

```diff
@@ -242,7 +256,7 @@ namespace ts {
                 fileExists: host.fileExists(fileOrDirectoryPath),
                 directoryExists: host.directoryExists(fileOrDirectoryPath)
             };
-            if (fsQueryResult.directoryExists || hasEntry(parentResult.directories, baseName)) {
+            if (fsQueryResult.directoryExists || hasEntry(parentResult.sortedAndCanonicalizedDirectories, getCanonicalFileName(baseName))) {

```

Comment:
Should this be a binary search too?

Reply:
It looks like `hasEntry` is just a search over the array that calls `getCanonicalFileName` on all values (including the entry being searched for, over and over...), so it seems like it could be.

Reply:
Oh, duh, you changed `hasEntry` above.